tests: net: socket: tls_ext move to new ztest API

hunterhu3000 opened the pull request:
Move net/socket/tls_ext tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

nashif commented on tests/net/socket/tls_ext/src/main.c:
```diff
@@ -399,22 +399,22 @@ static void test_common(int peer_verify)
 }
 
-static void test_tls_peer_verify_required(void)
+ZTEST(tls_socket_api_extension_test_suite, test_tls_peer_verify_required)
```
we do not need _test_suite suffix here.

hunterhu3000 replied:
Updated. Thanks to review.

hunterhu3000 committed:
tests: net: socket: tls_ext move to new ztest API

Move net/socket/tls_ext tests to use new ztest API
TEST=west build -p -b native_posix tests/net/socket/tls_ext

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
@nashif  thanks to review and +1.
